lib/libutee/tee_api_objects.c

```diff
@@ -501,21 +501,34 @@ TEE_Result TEE_RenamePersistentObject(TEE_ObjectHandle object,
 {
 	TEE_Result res;
 
-	if (object == TEE_HANDLE_NULL)
-		return TEE_ERROR_ITEM_NOT_FOUND;
+	if (object == TEE_HANDLE_NULL) {
+		res = TEE_ERROR_ITEM_NOT_FOUND;
+		goto err;
+	}
 
-	if (newObjectID == NULL)
-		return TEE_ERROR_BAD_PARAMETERS;
+	if (!newObjectID) {
+		res = TEE_ERROR_BAD_PARAMETERS;
+		goto err;
+	}
 
-	if (newObjectIDLen > TEE_OBJECT_ID_MAX_LEN)
-		TEE_Panic(0);
+	if (newObjectIDLen > TEE_OBJECT_ID_MAX_LEN) {
+		res = TEE_ERROR_BAD_PARAMETERS;
+		goto err;
+	}
 
 	res = utee_storage_obj_rename(object, newObjectID, newObjectIDLen);
 
-	if (res != TEE_SUCCESS && res != TEE_ERROR_ACCESS_CONFLICT)
-		TEE_Panic(0);
+	if (res == TEE_SUCCESS)
+		goto out;
+err:
+	if (res == TEE_ERROR_ACCESS_CONFLICT ||
```

Comment:
As Jens noted in #355, `If you added res == TEE_SUCCESS here you could get rid of the if just above`